src/Raven.Studio/typescript/viewmodels/database/cmpXchg/editCmpXchg.ts

```diff
@@ -56,203 +192,182 @@ class editCmpXchg extends viewModelBase {
                 }]
         });
 
-        this.valueText.extend({
-            required: true,
-            aceValidation: true
+        this.valueEditor().contentText.extend({
+            required: true
         });
-    }
-
-    private initializeObservables(): void {
-        this.dirtyFlag = new ko.DirtyFlag([this.valueText, this.key], false, jsonUtil.newLineNormalizingHashFunction); 
-          
-        this.value.subscribe(value => {
-            if (!_.isUndefined(value)) {
-                const valueText = this.stringify(value);
-                this.valueText(valueText);
+        
+        this.metadataEditor().contentText.extend({
+            required: {
+                onlyIf: () => this.hasMetadata()
             }
         });
-        
-        this.isSaveEnabled = ko.pureComputed(() => {
-            const isSaving = this.isSaving();
-            const isDirty = this.dirtyFlag().isDirty();
 
-            return !isSaving && isDirty;
-        });         
+        this.globalValidationGroup = ko.validatedObservable({
+            key: this.key,
+            valueText: this.valueEditor().contentText,
+            metadataText: this.metadataEditor().contentText
+        });
     }
 
-    updateNewlineLayout(unescapeNewline: boolean) {
-        const dirtyFlagValue = this.dirtyFlag().isDirty();
-        if (unescapeNewline) {
-            this.valueText(documentHelpers.unescapeNewlinesAndTabsInTextFields(this.valueText()));
-            this.valueEditor.getSession().setMode('ace/mode/json_newline_friendly');
-        } else {
-            this.valueText(documentHelpers.escapeNewlinesAndTabsInTextFields(this.valueText()));
-            this.valueEditor.getSession().setMode('ace/mode/raven_document');
-            this.formatValue();
-        }
+    attached() {
+        super.attached();
 
-        if (!dirtyFlagValue) {
-            this.dirtyFlag().reset();
-        }
+        popoverUtils.longWithHover($(".key-info"),
+            {
+                content: `<small>A unique identifier that is <strong>reserved accross the cluster</strong></small>`,
+                html: true,
+                placement: "bottom"
+            });
+        
+        popoverUtils.longWithHover($(".value-info"),
+            {
+                content: `<small>A Value that is associated with the Key</small>`,
+                html: true,
+                placement: "bottom"
+            });
+        
+        popoverUtils.longWithHover($(".metadata-info"),
+            {
+                content: `<ul class="margin-top margin-top-xs margin-right">
+                              <li><small>The Metadata is associated with the Key, similar to a document's metadata.</small></li>
+                              <li>
+                                  <small>Can be used to set the <strong>Compare Exchange item expiration.</strong></small><br>
+                                  <small>Set the <i>@expires</i> field to schedule expiration.</small><br>
```

Comment:
do we also need to enable anything in db config?

Reply:
No need to enable, it works
Existing configurations are:
![image](https://user-images.githubusercontent.com/17376974/103198733-bec79080-48f1-11eb-8f83-b86e9a241ade.png)